Scripts/BorrowingFrequency.sql

```diff
@@ -0,0 +1,15 @@
+-- 3. Borrowing Frequency using Window Functions
+--  Rank borrowers based on borrowing frequency.
+WITH BorrowerFrequency AS (
+    SELECT
+        BorrowerID,
+        COUNT(*) AS BorrowingFrequency,
+        ROW_NUMBER() OVER (ORDER BY COUNT(*) DESC) AS BorrowerRank

```

Comment:
Just a note, `ROW_NUMBER()`  assigns a unique rank to each borrower even if two borrowers have the same borrowing frequency it we want to make the same frequency to have the same rank and skip subsequent ranks, consider using `RANK()` or `DENSE_RANK()` instead.